### ISSUE TITLE
Create app or script to enhance user experience

### ISSUE THREAD

**Issue by subfuzion** [GoogleCloudPlatform/terraform-mean-cloudrun-mongodb#6]
Depends on #7 

Should prompt for configuration, like regions.

Updated:

- [ ] Collect configuration information (like regions, project name, etc.) to pass to Terraform.
- [ ] Use Cloud Build to deploy from local directory, GitHub repo, CDN, etc.
- [ ] Other build steps?
- [ ] Launch Terraform
- [ ] Run verification test (#12)?

**Comment by bleything** [comment]
Let's start a discussion about this so we can all be thinking about it while we continue working on the MVP. 

My initial thinking is that this is not a good use of our time, for a few reasons:

* the documentation should be a complete guide to setup and configuration
* terraform will already prompt for missing variable values
* the engineering effort required to create a robust script that wraps terraform is significant and (imo) and provides relatively little value vs. strong documentation

That said, I'm a bit uneasy committing either way right now. I think we should wait and see how #8 shakes out before we make any decisions about whether or not this is necessary or a good use of our limited time.

**Comment by subfuzion** [comment]
Agree with waiting to see how the MVP fleshes out first. This is a strawman proposal from the original project request. Per our offline discussion, I would favor going down the path of submitting patches to the [Atlas provider](https://github.com/mongodb/terraform-provider-mongodbatlas) before shelling out to external scripts.

One of the initial ideas was that this would be some kind of "pre-config" script, but that really won't help when APIs, such as [this one for determining regions](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#operation/returnAllCloudProviderRegions), requires that a project already exist.

I don't really think we want to create and manage any apps or scripts outside of Terraform. And so far, the only real use case I see at the moment for extending the Atlas provider is for fetching a list of valid GCP resources using the Atlas-specific names, including the ones eligible for free tier -- and this is something we probably can just ask the MongoDB folks to do.

**Comment by bleything** [comment]
Yeah, strong agree that the cleanest solution is a data source in the atlas provider. I mentioned this in #9 but I'm not sure that API endpoint actually does what we want it to do. I haven't been able to figure out how to call it from curl (to my great shame) and so I can't confirm 100% but it looks like it is designed to return the atlas regions that support a given cluster tier.

> I don't really think we want to create and manage any apps or scripts outside of Terraform. 

💯

**Comment by subfuzion** [comment]
Reopening this because although it would be nice to do everything with Terraform, it seems the original idea is still a good one to consider for enhancing the experience.
